source/m2mtimerpimpl.cpp

```diff
@@ -73,19 +73,19 @@ M2MTimerPimpl::M2MTimerPimpl(M2MTimerObserver& observer)
   _type(M2MTimerObserver::Notdefined),
   _intermediate_interval(0),
   _total_interval(0),
+  _still_left(0),
   _status(0),
   _dtls_type(false)
 {
     ns_hal_init(NULL, MBED_CLIENT_EVENT_LOOP_SIZE, NULL, NULL);
     eventOS_scheduler_mutex_wait();
     if (_tasklet_id < 0) {
-        _tasklet_id = eventOS_event_handler_create(tasklet_func, MBED_CLIENT_TIMER_EVENT);
+        _tasklet_id = eventOS_event_handler_create(tasklet_func, 0 /*ARM_LIB_TASKLET_INIT_EVENT*/);

```

Comment:
ARM_LIB_TASKLET_INIT_EVENT is defined in net_interface.h which lives under sal-stack-nanostack component. Adding that component will cause dependency to IPv6 stuff and that's not what we want?